### PR TITLE
Euler's totient function returns integers

### PR DESCRIPTION
**Pull request by pelegm**
Including a relevant test.

### CODE REVIEW

**Comment by asmeurer** [comment]
totient is kind of an odd function. According to the error message, the input must be a positive integer for it to even make sense.

**Comment by pelegm** [comment]
What's so odd about it? Many functions are defined on positive integers only...

**Comment by asmeurer** [comment]
The point is, shouldn't totient.is_integer just always return True?

**Comment by pelegm** [comment]
> The point is, shouldn't totient.is_integer just always return True?

I completely understand what you're saying. However, I was following a convention, introduced by @skirpichev; see for example #8490.

According to this convention, if the function is possibly not defined, then nothing is known about its output. So, in this case, if `x` is only known to be real, `totient(x)` is not known to be an integer, since it may be **undefined** as well.

This is a decision, and not one that I've made. Perhaps no one has made it yet, and we can open it for discussion. However, it must be consistent.

**Comment by smichr** [comment]
Could you rebase this, please?

**Comment by pelegm** [comment]
Sure, I'll do that.

**Comment by smichr** [comment]
And the fuzzy logic looks right to me. If you know it's undefined then the answer is False; if you don't know if it's undefined then the answer is None.

**Comment by pelegm** [comment]
@smichr, rebased.

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 from .generate import sieve, primerange, nextprime
 from sympy.core import sympify
 from sympy.core.evalf import bitcount
-from sympy.core.numbers import igcd, Rational
+from sympy.core.logic import fuzzy_and
+from sympy.core.numbers import igcd, oo, Rational
```
Comment (smichr) [inline sympy/ntheory/factor_.py]: looks like oo snuck back in during rebase. Is that needed in the file?

**Comment by pelegm** [comment]
@smichr, fixed rebase.

**Pull request merged**